specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json

```diff
@@ -5201,6 +5969,65 @@
         }
       }
     },
+    "SqlStoredProcedureResource": {
+      "type": "object",
+      "description": "Cosmos DB SQL storedProcedure resource object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Name of the Cosmos DB SQL storedProcedure"
+        },
+        "body":{
+          "type": "string",
+          "description": "Body of the Stored Procedure"
+        }
+      },
+      "required": [
+        "id"
+      ]
+    },
+    "SqlUserDefinedFunctionResource": {
+      "type": "object",
+      "description": "Cosmos DB SQL userDefinedFunction resource object",
+      "properties": {
+        "id": {
```

Comment:
Please make this more descriptive, id will be confused with the resource id field. For example, it could be called 'displayName' if this fits your scenario.

Reply:
Hi, @ryansbenson for all our resources we have used "id". Using "id" is the naming convention we have followed for all our resources. '_rid' is an internal property so the user never specifies it. This is strictly tied to our data plane contract as well so this needs to remain "id""

Reply:
Honestly this is really confusing but lets not address the greater issue in this PR.